tests/components/nx584/test_binary_sensor.py

```diff
@@ -73,79 +79,81 @@ def test_setup_full_config(self, mock_nx, mock_watcher):
             ]
         )
         assert add_entities.called
         assert nx584_client.Client.call_count == 1
         assert nx584_client.Client.call_args == mock.call("http://foo:123")
         assert mock_watcher.called
 
-    def _test_assert_graceful_fail(self, config):
+    @staticmethod
+    async def _test_assert_graceful_fail(hass, config):
         """Test the failing."""
-        assert not setup_component(self.hass, "nx584", config)
-
-    def test_setup_bad_config(self):
+        assert not await async_setup_component(hass, "nx584", config)
+
+    @pytest.mark.parametrize(
+        "config",
+        [
+            ({"exclude_zones": ["a"]}),
+            ({"zone_types": {"a": "b"}}),
+            ({"zone_types": {1: "notatype"}}),
+            ({"zone_types": {"notazone": "motion"}}),
+        ],
+    )
+    async def test_setup_bad_config(self, hass, config):
         """Test the setup with bad configuration."""
-        bad_configs = [
-            {"exclude_zones": ["a"]},
-            {"zone_types": {"a": "b"}},
-            {"zone_types": {1: "notatype"}},
-            {"zone_types": {"notazone": "motion"}},
-        ]
-        for config in bad_configs:
-            self._test_assert_graceful_fail(config)
-
-    def test_setup_connect_failed(self):
-        """Test the setup with connection failure."""
-        nx584_client.Client.return_value.list_zones.side_effect = (
-            requests.exceptions.ConnectionError
-        )
-        self._test_assert_graceful_fail({})
-
-    def test_setup_no_partitions(self):
-        """Test the setup with connection failure."""
-        nx584_client.Client.return_value.list_zones.side_effect = IndexError
-        self._test_assert_graceful_fail({})
-
-    def test_setup_version_too_old(self):
+        await self._test_assert_graceful_fail(hass, config)
+
+    @pytest.mark.parametrize(
+        "exception_type",
+        [
+            pytest.param(requests.exceptions.ConnectionError, id="connect_failed"),
+            pytest.param(IndexError, id="no_partitions"),
+        ],
+    )
+    async def test_setup_with_exceptions(self, hass, exception_type):
+        """Test the setup handles exceptions."""
+        nx584_client.Client.return_value.list_zones.side_effect = exception_type
+        await self._test_assert_graceful_fail(hass, {})
+
+    async def test_setup_version_too_old(self, hass):
         """Test if version is too old."""
         nx584_client.Client.return_value.get_version.return_value = "1.0"
-        self._test_assert_graceful_fail({})
+        await self._test_assert_graceful_fail(hass, {})
 
-    def test_setup_no_zones(self):
+    @staticmethod
+    def test_setup_no_zones(hass):
         """Test the setup with no zones."""
         nx584_client.Client.return_value.list_zones.return_value = []
         add_entities = mock.MagicMock()
-        assert nx584.setup_platform(self.hass, {}, add_entities)
+        assert nx584.setup_platform(hass, {}, add_entities)
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6372789)